### PR TITLE
Ensure shell.nix uses same nixpkgs as the rest of the environment

### PR DESCRIPTION
**Pull request by pedropombeiro**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

fixes https://github.com/status-im/infra-ci/issues/6

### Summary

[comment]: # (Summarise the problem and how the pull request solves it)
This PR ensures a single source of truth for our pinned version of nixpkgs and ensures that shell.nix follows it.

status: ready

### CODE REVIEW

**Comment by pedropombeiro** [comment]
Interesting, the `android-e2e` build failed with:

```
00:14:13.833  Error: A non-recoverable condition has triggered.  Watchman needs your help!
00:14:13.833  The triggering condition was at timestamp=1560939059: inotify-add-watch(/home/jenkins/workspace/status-react_prs_windows_PR-8433/node_modules/@babel/helper-create-class-features-plugin/node_modules/@babel/core/lib/config/validation) -> The user limit on the total number of inotify watches was reached; increase the fs.inotify.max_user_watches sysctl
00:14:13.833  All requests will continue to fail with this message until you resolve
00:14:13.833  the underlying problem.  You will find more information on fixing this at
00:14:13.833  https://facebook.github.io/watchman/docs/troubleshooting.html#poison-inotify-add-watch
00:14:13.833  
00:14:13.833      at ChildProcess.<anonymous> (/home/jenkins/workspace/us-react_prs_android-e2e_PR-8434/node_modules/fb-watchman/index.js:207:21)
00:14:13.833      at ChildProcess.emit (events.js:182:13)
00:14:13.833      at maybeClose (internal/child_process.js:962:16)
00:14:13.833      at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
00:14:13.833  Emitted 'error' event at:
00:14:13.833      at Client.client.on.error (/home/jenkins/workspace/us-react_prs_android-e2e_PR-8434/node_modules/jest-haste-map/build/lib/WatchmanWatcher.js:103:10)
00:14:13.833      at Client.emit (events.js:182:13)
00:14:13.833      at ChildProcess.<anonymous> (/home/jenkins/workspace/us-react_prs_android-e2e_PR-8434/node_modules/fb-watchman/index.js:214:12)
00:14:13.833      at ChildProcess.emit (events.js:182:13)
00:14:13.833      at maybeClose (internal/child_process.js:962:16)
00:14:13.833      at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
```

Maybe watchman is needed after all?

File: default.nix [hunk 1]
```diff
@@ -1,16 +1,11 @@
...
-{ config ? { android_sdk.accept_license = true; },
-  pkgs ? import ((import <nixpkgs> { }).fetchFromGitHub {
-    owner = "status-im";
-    repo = "nixpkgs";
-    rev = "db492b61572251c2866f6b5e6e94e9d70e7d3021";
-    sha256 = "188r7gbcrxi20nj6xh9bmdf3lbjwb94v9s0wpacl7q39g1fca66h";
-    name = "nixpkgs-source";
-  }) { inherit config; },
+{ config ? null,
+  nixpkgs-bootstrap ? import ./nix/nixpkgs-bootstrap.nix { inherit config; },
```
Comment (jakubgs) [inline default.nix:3]: Nice, I like it's in one place now.

**Review by jakubgs — APPROVED** [review]

**Review by jakubgs — APPROVED** [review]
Pretty sure we can also remove `android/gradlew`.

**Review by rasom — APPROVED** [review]

**Review by mandrigin — APPROVED** [review]

**Comment by jakubgs** [comment]
Took the liberty to squash the changes before merging.

**Pull request merged**